lib/psql.js

```diff
@@ -410,18 +416,24 @@ var PSQL = function(connectionParams, poolParams, options) {
                 };
                 client.on('notice', noticeListener);
 
-                var errorHappened = false;
                 query.on('error', function() {
-                    errorHappened = true;
+                    client.removeListener('notice', noticeListener);
+                    done(options.destroyOnError);
+                });
+
+                client.once('maxRowSize', function (message) {
+                    pg.cancel(that.getConnectionConfig(), client, query);
+                    query.emit('error', new Error('Row too large, was ' + message.length + ' bytes'));
                 });
 
                 // NOTE: for some obscure reason passing "done" directly
                 //       as the listener works but can be slower
                 //      (by x2 factor!)
                 query.on('end', function() {
                     client.removeListener('notice', noticeListener);
-                    done(options.destroyOnError && errorHappened);
+                    done(false);
```

Comment:
It's going to work, if it receives a `falsy` value then the client won't be released. See: https://github.com/brianc/node-pg-pool/blob/master/index.js#L97-L106

So, if you prefer call it without any param then fine by me.